docs/DeveloperGuide.md

```diff
@@ -1,38 +1,516 @@
+
+1. Command completion on tab
+
+2. Displaying option descriptions on hover
+
+3. Contextual help menus
+
+![linereader](images/ui-linereader-enhancement.png)
+
+This allows us to overload options on a small number of commands to provide full functionality of the application. Developers can then extend its features without also the worry of finding a way for users to access those features easily.
+
+### Storage Component
+
+![Storage class diagram](images/StorageClassDiagram.png)
+
+<!-- TODO: Link storage API -->
+
+#### Overview
+The `Storage` class is responsible for managing storage operations for the BinBash application. It performs critical 
+functions such as loading and saving item data, handling corrupted files, and maintaining the integrity of the 
+application's persistent data.
+
+#### Key Responsibilities
+- **Loading Data**: The `loadData()` method reads from `items.txt` and constructs a list of `Item` objects.
+- **Saving Data**: The `saveToStorage(List<Item> itemList)` method writes the current state of `Item` objects to `items.txt`.
+- **Corruption Handling**: If data corruption is detected, `handleCorruptedFile()` attempts to recover by renaming the corrupted file and creating a new one.
+- **Data Parsing**: The class contains methods for parsing data from and to the storage format, specifically `parseLinesToItemList(ArrayList<String>)` and `generateStorageRepresentationOfSingleItem(Item)`.
+
+
+### Parser Component
+
+![ParserClassDiagram](images/ParserClassDiagram.png)
+
+API: [`Parser.java`](https://github.com/AY2324S2-CS2113T-T09-2/tp/blob/master/src/main/java/seedu/binbash/parser/Parser.java)
+
+The `Parser` component plays a vital role in interpreting user input and facilitating interaction with the application. It serves as the bridge between user commands and actionable operations within the system. Here's an overview of its functionalities:
+
+- **Command Interpretation**: The `Parser` analyzes user input to identify the corresponding commands and parameters, ensuring accurate command interpretation.
+
+- **Command Delegation**: Once user commands are identified, the `Parser` delegates command execution to the appropriate command handlers, ensuring seamless operation flow.
+
+- **Error Handling**: In cases of invalid or malformed input, the `Parser` provides informative error messages, guiding users towards correct command usage.
+
+- **Integration Support**: Designed with modularity in mind, the `Parser` facilitates integration with other system components, promoting extensibility and maintainability.
+
+By effectively parsing user commands and translating them into actionable tasks, the `Parser` enhances the overall usability and functionality of the application.
+
+Below shows the sequence diagram of a `Parser` parsing user input, to return the corresponding `Command`.
+
+![ParseSequenceDiagram](images/ParseSequenceDiagram.png)
```

Comment:
Consider using `hide footbox` to hide this portion
![Screenshot 2024-04-04 141337](https://github.com/nus-cs2113-AY2324S2/tp/assets/89649663/4c9d757c-bdd7-4cf5-b710-7a19e09d1346)